helm-frontend/src/components/MiniLeaderboardTables.tsx

```diff
@@ -0,0 +1,303 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useEffect, useState } from "react";
+import { ChevronUpDownIcon } from "@heroicons/react/24/solid";
+import type GroupsTable from "@/types/GroupsTable";
+import RowValue from "@/components/RowValue";
+import Schema from "@/types/Schema";
+import getSchema from "@/services/getSchema";
+
+interface Props {
+  groupsTables: GroupsTable[];
+  activeGroup: number;
+  ignoreHref?: boolean;
+  sortable?: boolean;
+  sortFirstMetric?: boolean;
+  modelsToFilter?: string[];
+  numModelsToAutoFilter?: number;
+  filteredCols?: any[];
+}
+
+export default function MiniLeaderboardTables({
+  groupsTables,
+  activeGroup,
+  ignoreHref = false,
+  sortable = true,
+  sortFirstMetric = true,
+  filteredCols = [],
+  modelsToFilter = [],
+  numModelsToAutoFilter = 0, // if non-zero, sets how many models to filter down to (ranked by first column)
+}: Props) {
+  const [activeSortColumn, setActiveSortColumn] = useState<number | undefined>(
+    sortFirstMetric ? 1 : undefined,
+  );
+  const [activeGroupsTable, setActiveGroupsTable] = useState<GroupsTable>({
+    ...groupsTables[activeGroup],
+  });
+  const [sortDirection, setSortDirection] = useState<number>(1);
+  const [filteredModels, setFilteredModels] =
+    useState<string[]>(modelsToFilter);
+
+  interface HeaderValueObject {
+    value: string;
+  }
+
+  function truncateHeader(value: string): string {
+    if (value.length > 30) {
+      return value.substring(0, 27) + "...";
```

Comment:
Can we just do `text-overflow: ellipses` or does that not work?

Reply:
doesn't work as is